src/libsyntax/ast.rs

```diff
@@ -921,6 +921,15 @@ pub enum UnsafeSource {
     UserProvided,
 }
 
+/// A constant (expression) embedded inside a type, an expression,
+/// or used to define the discriminant value of an enum variant.
```

Comment:
Maybe we should call this an `AnonymousConst` or something like that? This would address @petrochenkov's concern about const vs static items to some extent, since they are clearly not anonymous.

Reply:
Yeah, `Const` seems too generic for this thing that has a very specific technical purpose rather than represents a constant expressions in general.

Reply:
I was thinking of `ty::Const` when I picked the name, TBH, but that doesn't make as much sense in the AST/HIR.

I guess `AnonConst` would work pretty well, thanks!